### PR TITLE
Add getCapability and GetManufacturer

### PR DESCRIPTION
**Pull request by DenisKarch**
NVReadValue consumes a location and a length and tries to read length
bytes from the requested location in NVRAM.

ReadEKCert reads the EKCert at the default location in NVRAM.

### CODE REVIEW

File: tpm/commands.go [hunk 1]
```diff
@@ -143,6 +143,17 @@ func getPubKey(rw io.ReadWriter, keyHandle tpmutil.Handle, ca *commandAuth) (*pu
 	return &pk, &ra, ret, nil
 }
 
+// getCapability reads the requested capability and sub-capability from NVRAM
+func getCapability(rw io.ReadWriter, cap, subcap uint32) ([]byte, error) {
+	var b []byte
+	in := []interface{}{cap, uint32(4), subcap} // tcsd requires capabilities to be of size 4, cannot find any mention of this in the spec
```
Comment (ericchiang) [inline tpm/commands.go]: This seems to contradict the existing use of ordGetCapability 

https://github.com/google/go-tpm/blob/7a7fe86fbbf203a562400a8a0dbe63a93ae02059/tpm/tpm.go#L46

Any thoughts why this is different?

File: tpm/tpm.go [hunk 2]
```diff
@@ -754,6 +754,11 @@ func ReadPubEK(rw io.ReadWriter) ([]byte, error) {
 	return tpmutil.Pack(pk)
 }
 
+// GetManufacturer returns the manufacturer ID
+func GetManufacturer(rw io.ReadWriter) ([]byte, error) {
+	return getCapability(rw, capProperty, tpmCapPropManufacturer)
```
Comment (ericchiang) [inline tpm/tpm.go]: nit: you probably don't need getCapability as a different method if its only used here.

File: tpm/commands.go (hunk 1 above)
Comment (DenisKarch) [reply tpm/commands.go]: It actually doesn't contradict it. The existing use packs the subcapability into a byte slice whereas I have the length and the actual subcap seperate (length+rawBytes). They are both encoded the same way though technically the existing call matches the spec more closely.

File: tpm/tpm.go (hunk 2 above)
Comment (DenisKarch) [reply tpm/tpm.go]: Although it is only used here it can be used to get a variety of other fields so I figured it would be best to abstract now.

Comment (ericchiang) [reply tpm/tpm.go]: Can you re-use this for GetKeys then?

Comment (DenisKarch) [reply tpm/tpm.go]: Done

**Comment by DenisKarch** [comment]
cc @tmroeder

**Comment by tmroeder** [comment]
LGTM.

My only comment is that the git commit should use the imperative form of the verb rather than the past:

"Add getCapability and GetManufacturer"

**Comment by awly** [comment]
Looks like there's a merge conflict

**Comment by DenisKarch** [comment]
Did a rebase, should work now.

**Pull request merged**